api/pkg/service/catalog/syncer.go

```diff
@@ -282,6 +283,28 @@ func (s *syncer) updateResourceTags(
 	}
 }
 
+func (s *syncer) updateResourcePlatforms(
+	txn *gorm.DB, log *zap.SugaredLogger,
+	res *model.Resource, platforms []string) {
+
+	if len(platforms) == 0 {
+		return
+	}
+
+	// clean existing platforms
+	txn.Unscoped().Where(&model.ResourcePlatform{ResourceID: res.ID}).Delete(&model.ResourcePlatform{})
```

Comment:
do we need to clean the resources before adding?

Reply:
I think `FirstOrCreate` will create record only if it is missing, so what if we don't clean then?

Reply:
In case if we don't clean platforms, new platforms will be just added to the existing ones (when it's necessary). This doesn't cover the use case when some platform was removed from the list of supported platforms in the resource.
Clean up of the existing records and addition of the new list would represent the actual data from the resource.